nsqd/lookup.go

```diff
@@ -196,3 +228,38 @@ func (n *NSQD) lookupdHTTPAddrs() []string {
 	}
 	return lookupHTTPAddrs
 }
+
+type state struct {
+	Paused bool `json:"paused"`
+}
+
+func (s *state) MarshalToJson() []byte {
+	if s == nil {
+		return nil //almost impossible to happen, just return nil
+	}
+	data, err := json.Marshal(s)
+	if err != nil {
+		return nil //almost impossible to happen, just return nil
+	}
+	return data
+}
+
+func getTopicState(topic *Topic) *state {
+	if topic == nil {
+		return &state{Paused: false}
+	}
+	s := state{
+		Paused: topic.IsPaused(),
+	}
+	return &s
+}
+
+func getChannelState(channel *Channel) *state {
+	if channel == nil {
+		return &state{Paused: false}
+	}
+	s := state{
+		Paused: channel.IsPaused(),
+	}
+	return &s
+}
```

Comment:
For a few reasons, we should be able to drop all these "helpers" after we make the changes discussed here https://github.com/nsqio/go-nsq/pull/305#pullrequestreview-467949543